### PR TITLE
command/import: fix error when given/configured provider != implied provider type name

### PR DESCRIPTION
**Pull request by mildwonkey**
This issue only presented when both of the following are true:

- The resource's provider does not match the implied provider (such as "test-beta" provider for "test_instance" ).
           **AND**
- The config did not include the implied provider at all ("test" was not initialized, because it's not used, only implied).

Previously, the `NodeAbstractResource` would be associated with the implied provider - even if the provider was explicitly configured - then fail when terraform discovered that the implied provider wasn't initialized. Now, the explicit provider is attached (when set), so it does not try to evaluate the implied provider. 

In the future, it would be nice to implement the following (either together or separately, but in this order):
- use the `provider` from the configuration, removing the need for `-provider` on the command line
- Either remove the `-provider` flag entirely, or fail if the `-provider` given on the CLI does not match the configured provider (it may turn out to be more complicated than this, when you consider aliases and inheritance) 

---
I will squash this PR before merging, but chose to preserve my first attempt at fixing this bug for now because it was a fascinating journey :)

### CODE REVIEW

File: command/import.go [hunk 1]
```diff
@@ -178,6 +178,9 @@ func (c *ImportCommand) Run(args []string) int {
 		// We assume the same module as the resource address here, which
 		// may get resolved to an inherited provider when we construct the
 		// import graph inside ctx.Import, called below.
+		if rc != nil && rc.ProviderConfigRef != nil {
+			providerAddr = rc.ProviderConfigAddr().Absolute(addr.Module)
+		}
 		providerAddr = resourceRelAddr.DefaultProviderConfig().Absolute(addr.Module)
```
Comment (jbardin) [inline command/import.go]: `providerAddr` seems to be re-assigned immediately after the conditional block here.

File: command/import.go [hunk 2]
```diff
@@ -178,6 +178,9 @@ func (c *ImportCommand) Run(args []string) int {
 		// We assume the same module as the resource address here, which
 		// may get resolved to an inherited provider when we construct the
 		// import graph inside ctx.Import, called below.
+		if rc != nil && rc.ProviderConfigRef != nil {
+			providerAddr = rc.ProviderConfigAddr().Absolute(addr.Module)
```
Comment (apparentlymart) [inline command/import.go]: In the `terraform import` docs for `-provider` it currently states the following:

> * `-provider=provider` - Specified provider to use for import. The value should be a provider alias in the form `TYPE.ALIAS`, such as "aws.eu". This defaults to the normal provider based on the prefix of the resource being imported. You usually don't need to specify this.

At the very least we should probably update the statement about the default to say that it consults the `provider` argument in the configuration.

However, I was also pondering whether we should reword this to be more definitive about the fact that this argument is now, in a sense, deprecated:

> `-provider=provider` - **Deprecated** Override the provider configuration to use when importing the object. By default, Terraform uses the provider specified in the configuration for the target resource, and that is the best behavior in most cases.

What do you think? :thinking:

File: command/import.go (hunk 1 above)
Comment (mildwonkey) [reply command/import.go]: 🤦‍♀ thank you

File: command/import.go (hunk 2 above)
Comment (mildwonkey) [reply command/import.go]: I like it! I've updated the cli usage and the website docs.
